### ISSUE TITLE
Travis-CI reports success even if Backdrop was not installed (Installation requirements problem: Missing Config module.)

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1165]
This is on a fresh installation:

```
Required modules    Required modules not found.
The following modules are required but were not found. Move them
into the appropriate modules subdirectory, such as /modules. Missing
modules: Config
```

![backdrop-installation-missing_config_module](https://cloud.githubusercontent.com/assets/2423362/9830561/a15a1d44-5976-11e5-9a9f-9802c34dda1c.png)

I've checked and the /core/modules/config folder is there and the folder contents seem to be in order. This is from the latest 1.1.x branch.

**Comment by klonos** [comment]
...weird how we didn't have any tests failing on this one.

**Comment by klonos** [comment]
...doesn't happen in 1.1.3.

**Comment by sutibun** [comment]
I can't be 100% sure but I think that's the error message I received when I installed backdrop the first time. I didn't know what to do so I left. Took some time before I tried playing with backdrop (somewhat recently)

The YouTube meeting videos got me back.

**Comment by quicksketch** [comment]
This regression was caused by #1110.

> ...weird how we didn't have any tests failing on this one.

This is an issue with our install.sh script. Apparently what was happening was that Travis failed to install Backdrop. Then when the run-tests.sh script ran, it returned nothing. Which Travis took to be an "All tests are passing". :stuck_out_tongue: 

The immediate problem was fixed by https://github.com/backdrop/backdrop/commit/91a6eaef4821a4928ad1a1edaa1eac9c2f1d9fb4, which made it so Backdrop could find its config.module again.

> ...doesn't happen in 1.1.3.

Yeah, fortunately this issue had only been present for the last 2 days since I merged #1110.

**Comment by docwilmot** [comment]
Didn't this mean we shouldn't trust any tests which ran over the last 2 days? Is there a way to rerun tests?

**Comment by quicksketch** [comment]
> Didn't this mean we shouldn't trust any tests which ran over the last 2 days? Is there a way to rerun tests?

I can restart them from Travis-CI. Anyone that made a PR can force push to it to restart the test as well. For all code that has been committed, we know it's okay as long as tests are passing currently (which they are).

**Comment by quicksketch** [comment]
I filed a PR at https://github.com/backdrop/backdrop/pull/1044 that improves our run-tests.sh script so that it will return a value of 1 (in Linux a return value other than 0 means there was an error) if Backdrop was not installed properly.

I actually screwed this up the first time, and the tests actually forcibly failed every time, so we know that this will bail out properly (see [this build for the failure](https://travis-ci.org/backdrop/backdrop/builds/80050532)).

This [fixed build](https://travis-ci.org/backdrop/backdrop/builds/80051788) now should include the change to prevent Travis from saying all tests passed when none were run in the first place.
